example/bluetooth_provisioning/lib/bluetooth_provisioning_flow.dart

```diff
@@ -37,50 +38,63 @@ class _BluetoothProvisioningFlowState extends State<BluetoothProvisioningFlow> {
     _onNextPage();
   }
 
-  void _onNameDeviceScreen(String ssid, String? psk) {
+  void _onWifiCredentials(String ssid, String? psk) async {
     final viewModel = Provider.of<BluetoothProvisioningFlowViewModel>(context, listen: false);
-    viewModel.setWifiCredentials(ssid: ssid, psk: psk);
-    _onNextPage();
+    try {
+      setState(() {
+        _isLoading = true;
+      });
+      await viewModel.writeConfig(ssid: ssid, psk: psk);
+      _onNextPage();
+    } catch (e) {
+      if (mounted) {
+        showErrorDialog(context, title: 'Failed to write config', error: e.toString());
+      }
+    } finally {
+      setState(() {
+        _isLoading = false;
+      });
+    }
   }
 
   @override
   Widget build(BuildContext context) {
     return Consumer<BluetoothProvisioningFlowViewModel>(
       builder: (context, viewModel, child) {
         return Scaffold(
-          appBar: AppBar(), // TODO: custom back button
+          appBar: AppBar(),
           body: SafeArea(
-            child: PageView(
-              controller: _pageController,
-              physics: NeverScrollableScrollPhysics(),
+            child: Stack(
               children: [
-                IntroScreenOne(handleGetStartedTapped: _onNextPage),
-                IntroScreenTwo(handleNextTapped: _onNextPage),
-                BluetoothScanningScreen(onDeviceSelected: _onDeviceConnected),
-                if (viewModel.connectedDevice != null)
-                  PairingScreen(
-                    connectedDevice: viewModel.connectedDevice!,
-                  ),
-                if (viewModel.connectedDevice != null)
-                  ConnectedBluetoothDeviceScreen(
-                    handleGoToNameDeviceScreen: _onNameDeviceScreen,
-                    robot: viewModel.robot,
-                    robotPart: viewModel.mainRobotPart,
-                    connectedDevice: viewModel.connectedDevice!,
-                  ),
-                if (viewModel.ssid != null && viewModel.connectedDevice != null)
-                  NameConnectedDeviceScreen(
-                    ssid: viewModel.ssid!,
-                    passkey: viewModel.psk,
-                    connectedPeripheral: viewModel.connectedDevice!,
-                  ),
-                if (viewModel.connectedDevice != null)
-                  CheckConnectedDeviceOnlineScreen(
-                    handleSuccess: _onNextPage, // TODO: diff handling tbh
-                    viam: viewModel.viam,
-                    robot: viewModel.robot,
-                    connectedDevice: viewModel.connectedDevice!,
+                Opacity(
+                  opacity: _isLoading ? 0.0 : 1.0,
```

Comment:
don't love this but the pageview not being in the hierarchy at some point can break things, can re-visit